rdflib/plugins/sparql/algebra.py

```diff
@@ -795,6 +798,433 @@ def translateQuery(q, base=None, initNs=None):
     return Query(prologue, res)
 
 
+class ExpressionNotCoveredException(Exception):
+    pass
+
+
+def translateAlgebra(query_algebra: Query = None):
```

Comment:
Not sure under what circumstance it makes sense for this to be null, because if it is null `traverse(query_algebra.algebra, visitPre=sparql_query_text)` will fail.

Reply:
You are right. I will correct it.

Reply:
Not too serious, just saw the error in mypy and thought I would let you know in case I was missing something.